fastlane/Fastfile

```diff
@@ -428,6 +429,87 @@ platform :android do
     )
   end
 
+  desc 'Builds a prototype build and uploads it to S3'
+  lane :build_and_upload_prototype_build do
+    UI.user_error!("'BUILDKITE_ARTIFACTS_S3_BUCKET' must be defined as an environment variable.") unless ENV['BUILDKITE_ARTIFACTS_S3_BUCKET']
+
+    prototype_build_type = 'debugProd'
+
+    gradle(
+      task: 'assemble',
+      build_type: prototype_build_type,
+      properties: {
+        'skipSentryProguardMappingUpload' => true
+      }
+    )
+
+    prototype_build_comments = lane_context[SharedValues::GRADLE_ALL_APK_OUTPUT_PATHS].map do |apk_path|
+      upload_path = upload_to_s3(
+        bucket: 'a8c-apps-public-artifacts',
+        key: "pocketcasts-#{get_app_key(apk_path: apk_path)}-prototype-build-#{generate_prototype_build_number}.apk",
+        file: apk_path,
+        skip_if_exists: true
+      )
+
+      install_url = "#{PROTOTYPE_BUILD_DOMAIN}/#{upload_path}"
+
+      prototype_build_details_comment(
+        app_display_name: get_app_display_name(apk_path: apk_path),
+        download_url: install_url,
+        metadata: {
+          'Build Type': prototype_build_type
+        },
+        fold: true
+      )
+    end
+
+    unless ENV['BUILDKITE_PULL_REQUEST'].nil?
```

Comment:
Thanks! Double check: it's `next`, not `return`?

Reply:
[It's `next` indeed](https://docs.fastlane.tools/advanced/lanes/#stop-executing-a-lane-early) (FYI because lanes are actually `do…end` _blocks_—that are `yield`-ed by _fastlane_'s lane runner coordinating which lane to run—and not bare ruby functions from which you'd `return)

Reply:
Gotcha, thanks! TIL

Reply:
After rethinking this suggestion, I've decided to propose not running "Prototype Builds" job if it's not for PR: 
e99b38b71d87f1bb293d2414eb0c8459ac8264f9

We already verify build correctness with "Assemble release APK" and instrumentation tests jobs, there's no need to assemble the apk one more time.

Reply:
Good point, makes sense to me 👍